source/extensions/clusters/logical_dns/logical_dns_cluster.cc

```diff
@@ -101,7 +103,14 @@ void LogicalDnsCluster::startResolve() {
         active_dns_query_ = nullptr;
         ENVOY_LOG(trace, "async DNS resolution complete for {} details {}", dns_address_, details);
 
+        std::chrono::milliseconds jitter(0);
+        if (dns_jitter_max_ms_.count() != 0) {
+          jitter = std::chrono::milliseconds(random_.random()) % dns_jitter_max_ms_;
+        }
         std::chrono::milliseconds final_refresh_rate = dns_refresh_rate_ms_;
+        if (jitter < final_refresh_rate) {
+          final_refresh_rate -= jitter;
```

Comment:
IMHO it is less intuitive to have a negative jitter.
Is there a clear reason why this shouldn't be positive?

Reply:
This makes sure that `final_refresh_rate` will always be positive. I'm not sure what you mean by "negative jitter"

Reply:
Or do you mean why I'm subtracting? In that case, I feel like the refresh rate should be at most the ttl. For example, if the ttl is 5s, we would want to refresh before the 5s is up to avoid having stale entries.

Reply:
If the refresh rate is set to duration X, then the actual refresh rate (with jitter) will always be >= X.

I've only seen cases where the jitter is either positive (added to some baseline) or symmetrical around the baseline (i.e., +/- jitter).

Reply:
sounds good